inst/private/assert_pythonic_and_sympy.m

```diff
@@ -0,0 +1,160 @@
+
+  minsympyver = '1.2';
+
+  if (nargin < 1)
+    verbose = false;
+  end
+
+  if (verbose)
+    disp ('')
+    disp ('Symbolic package diagnostics')
+    disp ('============================')
+    disp ('')
+    disp ('Python and SymPy are needed for most features of the Symbolic package.')
+    disp ('')
+    disp ('You seem to have loaded the Pythonic package; we will try to use it')
+    disp ('instead of directly calling a Python interpreter.  You can override')
+    disp ('this using "sympref" or by unloading Pythonic:')
+    disp ('    pkg unload pythonic')
+    disp ('or')
+    disp ('    sympref ipc popen2')
+    disp ('    sympref reset')
+    disp ('')
+    disp ('Attempting to run "py.str(''Python says hello'')":')
+    disp ('')
+  end
+
+  try
+    output = py.str('Python says hello');
+    if (verbose)
+      output
+    end
+    assert (strcmp (strtrim (char (output)), 'Python says hello'))
+  catch
+    if (~ verbose)
+      error ('OctSymPy:nopython', ...
+             ['Cannot use the Pythonic "py" command\n' ...
+              '    Try "sympref diagnose" for more information.']) ...
+    end
+    disp ('')
+    disp ('Unfortunately, that command failed!')
+    disp ('We expected to see "output = [Python object of type str] ... hello"')
+    disp ('')
+    disp ('  * Are you using the most recent Pythonic?')
+    disp ('    check: https://gitlab.com/mtmiller/octave-pythonic')
+    disp ('')
+    disp ('  * Please try unloading "pythonic" or using "sympref ipc" as above.')
+    disp ('')
+    return
+  end
+
+  if (verbose)
+    disp ('')
+    disp ('Good, Python ran correctly.')
+    disp ('')
+    disp ('Python version')
+    disp ('--------------')
+    disp ('')
+    pyversion ()
+  else
+    pyver = pyversion ();
+  end
+
+  if (verbose)
+    disp ('')
+    disp ('SymPy Python Library')
+    disp ('--------------------')
+    disp ('')
+    disp ('SymPy is a Python library used by Symbolic for almost all features.')
+    disp ('')
+    disp ('Attempting to run "py.sympy.__version__":')
+    disp ('')
+  end
+
+  try
+    output = py.sympy.__version__;
+    if (verbose)
+      output
+    end
+  catch
+    if (~ verbose)
+      error ('OctSymPy:nosympy', ...
+            ['Python cannot import SymPy: have you installed SymPy?\n' ...
+             '    Try "sympref diagnose" for more information.'])
+    end
+    disp ('')
+    disp ('Unfortunately we cannot import sympy.')
+    disp ('')
+    disp ('  * Do you have SymPy installed?  If not, please try to install it and')
+    disp ('    try again.')
+    disp ('')
+    disp ('  * If you do have SymPy installed, maybe it''s installed for a different')
+    disp ('    Python environment than the one Pythonic is linked to?  Please try')
+    disp ('    without Pythonic as described above.')
+    return
+  end
+
+  spver = strtrim (char (output));
+
+  if (verbose)
+    disp ('')
+    fprintf ('SymPy must be at least version %s; you have version "%s".\n', ...
```

Comment:
This output is slightly different from the original.

Without pythonic:

```
SymPy must be at least version 1.2; you have version 1.4.

Good, a working version of SymPy is installed.
```
With pythonic:

```
SymPy must be at least version 1.2; you have version "1.4".
Good, a working version of SymPy is installed.
```